File: LeeC/088_MergeSortedArray/Solution.py



class Solution:
    
    def merge(self, nums1, m, nums2, n):
    
        while m > 0 and n > 0:
            
            if nums1[m-1] >= nums2[n-1]:
                
                nums1[m+n-1] = nums1[m-1]
                m -= 1
                
            else:
                
                nums1[m+n-1] = nums2[n-1]
                n -= 1
                
        if n > 0:
            nums1[:n] = nums2[:n]

num1 = [1,2,3]
num2 = [5,2,3,7]


s1 = Solution()
s1.merge(num1,len(num1),num2,len(num2))

File: LeeC/088_MergeSortedArray/Solution.java



class Solution {
    
    public void merge(int A[], int m, int B[], int n) {
        
        for(int a=0; a<A.length; a++)
            System.out.print(A[a] + "  ");
        System.out.println();       
 
        for(int b=0; b<B.length; b++)
            System.out.print(B[b] + "  ");
        System.out.println();

        int i=m-1, j=n-1, k=m+n-1;
        
        while (i>-1 && j>-1) A[k--]= (A[i]>B[j]) ? A[i--] : B[j--];
        
        while (j>-1) A[k--]=B[j--];
        
        for(int a=0; a<A.length; a++)
            System.out.print(A[a] + "  ");
        System.out.println();       
    }
     
    public static void main(String args[]){
        Solution s = new Solution();

        int[] nums1 = new int[]{1,2,3,0,0,0};
        int m = 3;
        int[] nums2 = new int[]{2,5,6};
        int n = 3;
        s.merge(nums1, m, nums2, n);

    }   
}
